cli/context/store/store.go

```diff
@@ -259,9 +263,27 @@ func Export(name string, s Reader) io.ReadCloser {
 	return reader
 }
 
+const maxFileSize int64 = 1 << 20
+
 // Import imports an exported context into a store
 func Import(name string, s Writer, reader io.Reader) error {
-	tr := tar.NewReader(reader)
+	// need a buffered reader to read the header without advancing the reader
+	r := bufio.NewReader(reader)
```

Comment:
Decided to go with adding detection here; it adds a bit of complexity here, but by doing it here, the `Import()` signature remains unmodified. Also, by detecting, _piping_ a ZIP works (whereas previously it required a file as the extension was used).

Reply:
This and usage of the `DetectContentType` looks good to me.